packages/flutter/lib/src/widgets/heroes.dart

```diff
@@ -727,6 +727,28 @@ class HeroController extends NavigatorObserver {
     _maybeStartHeroTransition(route, previousRoute, HeroFlightDirection.pop, true);
   }
 
+  @override
+  void didStopUserGesture() {
+    if (navigator.userGestureInProgress)
+      return;
+
+    bool isStaleGesture(_HeroFlight flight) {
```

Comment:
This would be a good place to explain what a stale gesture is and why we care about them.